app-back/src/controllers/user.controller.js

```
@@ -0,0 +1,77 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { User } from '../models/user.model.js';
+import { userService } from '../services/user.service.js';
+import bcrypt from 'bcrypt';
+import { authController } from './auth.controller.js';
+
+const getAllActivated = async (req, res) => {
+  const users = await userService.getAllActivated();
+
+  res.send(users.map(userService.normalizedUser));
+};
+
+const changeName = async (req, res) => {
+  const { name } = req.body;
+  const userId = req.user.id;
+
+  if (!name || name.trim().length < 2) {
+    throw ApiError.badRequest('Name is invalid');
+  }
+
+  const user = await userService.changeName(userId, name.trim());
+
+  res.send({
+    user: userService.normalizedUser(user),
+  });
+};
+
+const changePassword = async (req, res) => {
+  const { oldPassword, newPassword, confirmPassword } = req.body;
+  const userId = req.user.id;
+
+  if (authController.validatePassword(newPassword)) {
+    return ApiError.badRequest();
```

When `validatePassword` returns an error message (a truthy value), you should `throw` the `ApiError` instead of returning it. The `catchError` utility is designed to catch thrown errors, but returning a value here will not send a proper response to the client.